x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/edit_field/edit_field_container.tsx

```diff
@@ -40,5 +52,47 @@ export const EditFieldContainer = React.memo(({ field, allFields }: Props) => {
     dispatch({ type: 'documentField.changeStatus', value: 'idle' });
   }, [dispatch]);
 
-  return <EditField form={form} field={field} allFields={allFields} exitEdit={exitEdit} />;
+  useEffect(() => {
+    isMounted.current = true;
+
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (isEditing) {
+      // Open the flyout with the <EditField /> content
+      addContent<EditFieldProps>({
+        id: 'mappingsEditField',
```

Comment:
The use of the id value is so that each consumer is independent. It can say at any moment: `removeContent('mappingsEditField')`, it does not know if that content is in the flyout or not. It just says to remove it. The "singleton" flyout then has the logic to deal with it and check if that content is actually the active content or not.

Without that, the consumer would say `closeFlyout()` ... but hey, maybe someone else changed the content of the flyout, who are you to dictate if the flyout should be closed! 😄